client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -126,11 +129,32 @@ tie.factory('PythonCodePreprocessorService', [
         while (true) {
           var codeToAdd = '';
           lastFunctionNameLocation = code.indexOf(
-            functionName, lastFunctionNameLocation);
+            functionNameWithoutParen, lastFunctionNameLocation);
           if (lastFunctionNameLocation === -1) {
             break;
           }
           if (lastFunctionNameLocation !== functionNameStart) {
+            var whitespaceCheckLocation = (
+              lastFunctionNameLocation + functionNameWithoutParen.length);
+            // We assume we found the matched function, but we might have found
+            // a similarly-named other function / variable. User-inputted code
+            // can be ... tricky.
+            var isMatchedFunction = true;
+            // We can safely assume there's a left paren because we matched it
+            // with the regular expression.
+            while (code[whitespaceCheckLocation] != '(') {
+              var isValidWhitespace = VALID_WHITESPACE_REGEX.test(
```

Comment:
I'm a little confused about why you need to do all this. We've already determined that the current location is "function name + possible spaces + open paren", right? If so, why not just scan until the next open paren?